server/src/main/java/org/elasticsearch/cluster/routing/allocation/NodeAllocationStatsProvider.java

```diff
@@ -23,17 +27,47 @@
 public class NodeAllocationStatsProvider {
     private final WriteLoadForecaster writeLoadForecaster;
 
-    public NodeAllocationStatsProvider(WriteLoadForecaster writeLoadForecaster) {
+    private volatile float indexBalanceFactor;
+    private volatile float shardBalanceFactor;
+    private volatile float writeLoadBalanceFactor;
+    private volatile float diskUsageBalanceFactor;
+
+    public record NodeAllocationAndClusterBalanceStats(
```

Comment:
This is `NodeAllocationStats` plus one more field. We could also directly reuse `NodeAllocationStats` but there is not much difference, IMO. (there would be still some nesting/unnesting code).